src/js/helpers/index.js

```diff
@@ -0,0 +1,17 @@
+import { useState, useRef, useEffect } from 'react';
+
+export const useStateWithPrevious = (initialValue) => {
```

Comment:
This _feels_ wrong to me, I think because it is basically going back to `componentDidUpdate` but with hooks. Can you describe the use case for this? (In other words, why do we need to know the previous value of these state vars that are in the dependency array of a useEffect? It seems like all we need to know is that the value has changed, and that is all we're using this to check for in `TableContainer`.)

Reply:
Sure. In [the react docs](https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state) they recommend this implementation. It does feel wrong but it's the official way to do it. 

Also:

> It’s possible that in the future React will provide a usePrevious Hook out of the box since it’s a relatively common use case.

The only reason we need this is because if we only check for if the value changed via the dependency array, it will fire off a duplicate request on initial render.

Reply:
Thanks!